src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -3613,10 +3611,10 @@ public Type visitTypeVar(TypeVar t, Void ignored) {
                     return to.head.withTypeVar(t);
                 }
                 if (allowUniversalTVars &&
-                        t.createdFromUniversalTypeVar &&
+                        !t.isValueProjection() &&
                         from.head.hasTag(TYPEVAR) &&
-                        ((TypeVar)from.head).referenceProjection != null &&
-                        t.equalsIgnoreMetadata(((TypeVar)from.head).referenceProjection)) {
+                        ((TypeVar)from.head).projection != null &&
+                        t.equalsIgnoreMetadata(((TypeVar)from.head).projection)) {
```

Comment:
and here, we should call `referencceProjection()` ?

Reply:
yep I agree